### DISCUSSION TITLE
Schema changes of adding index to JSON is not deployable

### DISCUSSION THREAD

**Discussion by renyuanz** [planetscale/discussion#202] [Bug Reports] [answered]
<img width="783" alt="截屏2022-05-28 下午3 59 02" src="https://user-images.githubusercontent.com/10615036/170816521-8c4a8f4e-1b90-4fd6-8ac8-d16bf217cd63.png">

I ran this `ALTER TABLE Contact ADD INDEX tags( (CAST(metadata ->'$.tags' AS CHAR(200) ARRAY)) );` in the development branch's console and create a deploy request, but the DR is not deployable without showing any instructions.

<img width="1347" alt="截屏2022-05-28 下午4 03 54" src="https://user-images.githubusercontent.com/10615036/170816589-1cdb55fd-2704-474f-9e26-aa73c08e8f87.png">

**Comment by jgreet** [comment, marked as answer]
Sorry for the trouble here. We have a two fixes for this problem underway.

First, the deploy-request page should now show the DDL to add the index and an appropriate error:

<img width="600" alt="CleanShot 2022-06-02 at 13 05 06@2x" src="https://user-images.githubusercontent.com/391331/171686239-0dc61c65-d0e6-4115-912e-e909324eea92.png">

Second, a future update will resolve the problem with deploying `ALTER` statements that add indexes for JSON columns using `array` in a `convert` or `cast` function. We expect that update to land in a couple of weeks and we'll update this thread when it does.
